sdk/storage/src/data_lake/clients/data_lake_client.rs

```diff
@@ -175,14 +106,18 @@ impl DataLakeClient {
         self.storage_client.storage_account_client().http_client()
     }
 
-    pub(crate) fn url(&self) -> &Url {
-        &self.url
+    pub(crate) fn url(&self) -> String {
```

Comment:
Could we make this a `&str`?

Looks like you even call `.as_str()` on the `String` returned from this, so returning a `&str` shouldn't be an issue.

Reply:
Will do

Reply:
Done